### PR TITLE
Docs: Add copy with headers docs

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Adds new copypaste options to the [`CopyPaste`](https://handsontable.com/docs/javascript-data-grid/api/copy-paste/#options) and [`copyPaste`](https://handsontable.com/docs/javascript-data-grid/api/options/#copypaste) API ref
- Adds new context menu items to the [Context menu](https://handsontable.com/docs/javascript-data-grid/context-menu/) page
- Cleans up the [Context menu with specific options](https://handsontable.com/docs/javascript-data-grid/context-menu/#context-menu-with-specific-options) table (other pieces of Context menu content still need refinement: https://github.com/handsontable/dev-handsontable/issues/16)
- Makes other minor changes in the content related to copy-paste

[skip changelog]

### CODE REVIEW

File: handsontable/src/dataMap/metaManager/metaSchema.js [hunk 1]
```diff
@@ -1211,18 +1211,19 @@ export default () => {
...
+     * | `columnsLimit`           | A number (default: `Infinity`)                     | The maximum number of columns that can be copied                                                                                                                                                    |
+     * | `rowsLimit`              | A number (default: `Infinity`)                     | The maximum number of columns that can be copied                                                                                                                                                    |
+     * | `pasteMode`              | `'overwrite'` \| `'shift_down'` \| `'shift_right'` | When pasting:<br>`'overwrite'`: overwrite the currently-selected cells<br>`'shift_down'`: move the currently-selected cells down<br>`'shift_right'`: move the currently-selected cells to the right |
+     * | `copyColumnHeaders`      | Boolean (default: `false`)                         | `true`: add the option to copy cells along with their nearest column headers                                                                                                                        |
```
Comment (budnix) [inline handsontable/src/dataMap/metaManager/metaSchema.js:1219]: WDYT about rephrasing descriptions to emphasize that they refer to the context menu? Something like: `true`: extend the context menu for an option that allows copy cells along with their nearest column headers.

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 2]
```diff
@@ -91,35 +88,39 @@ export class CopyPaste extends BasePlugin {
...
 
   /**
-   * Maximum number of columns than can be copied to clipboard using <kbd>**Ctrl**</kbd>/<kbd>**Cmd**</kbd> + <kbd>**C**</kbd>.
+   * The maximum number of columns than can be copied to the clipboard.
    *
    * @type {number}
    * @default Infinity
    */
   columnsLimit = Infinity;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:97]:
```suggestion
```
I guess the Team applies the rule of no new lines between class fields. Can you revert that?

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 3]
```diff
@@ -91,35 +88,39 @@ export class CopyPaste extends BasePlugin {
...
   /**
-   * Maximum number of rows than can be copied to clipboard using <kbd>**Ctrl**</kbd>/<kbd>**Cmd**</kbd> + <kbd>**C**</kbd>.
+   * The maximum number of rows than can be copied to the clipboard.
    *
    * @type {number}
    * @default Infinity
    */
   rowsLimit = Infinity;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:105]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 4]
```diff
@@ -91,35 +88,39 @@ export class CopyPaste extends BasePlugin {
...
+   * When pasting:
+   * - `'overwrite'`: overwrite the currently-selected cells
+   * - `'shift_down'`: move currently-selected cells down
+   * - `'shift_right'`: move currently-selected cells to the right
    *
    * @type {string}
    * @default 'overwrite'
    */
   pasteMode = 'overwrite';
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:116]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 5]
```diff
@@ -91,35 +88,39 @@ export class CopyPaste extends BasePlugin {
...
   /**
-   * UI container for the secondary focusable element.
+   * The UI container for the secondary focusable element.
    *
    * @type {HTMLElement}
    */
   uiContainer = this.hot.rootDocument.body;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:123]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 6]
```diff
@@ -128,6 +129,7 @@ export class CopyPaste extends BasePlugin {
    * @default false
    */
   #enableCopyColumnHeaders = false;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:132]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 7]
```diff
@@ -136,6 +138,7 @@ export class CopyPaste extends BasePlugin {
    * @default false
    */
   #enableCopyColumnGroupHeaders = false;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:141]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 8]
```diff
@@ -144,6 +147,7 @@ export class CopyPaste extends BasePlugin {
    * @default false
    */
   #enableCopyColumnHeadersOnly = false;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:150]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 9]
```diff
@@ -154,18 +158,21 @@ export class CopyPaste extends BasePlugin {
    * @type {'cells-only' | 'column-headers-only' | 'with-all-column-headers' | 'with-column-headers'}
    */
   #copyMode = 'cells-only';
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:161]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 10]
```diff
@@ -154,18 +158,21 @@ export class CopyPaste extends BasePlugin {
...
   /**
    * Flag that is used to prevent copying when the native shortcut was not pressed.
    *
    * @type {boolean}
    */
   #isTriggeredByCopy = false;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:168]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 11]
```diff
@@ -154,18 +158,21 @@ export class CopyPaste extends BasePlugin {
...
   /**
    * Flag that is used to prevent cutting when the native shortcut was not pressed.
    *
    * @type {boolean}
    */
   #isTriggeredByCut = false;
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:175]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 12]
```diff
@@ -179,13 +186,15 @@ export class CopyPaste extends BasePlugin {
     columnsLimit: () => this.columnsLimit,
     countColumnHeaders: () => this.hot.view.getColumnHeadersCount(),
   });
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:189]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 13]
```diff
@@ -179,13 +186,15 @@ export class CopyPaste extends BasePlugin {
...
   /**
    * Ranges of the cells coordinates, which should be used to copy/cut/paste actions.
    *
    * @private
    * @type {Array<{startRow: number, startCol: number, endRow: number, endCol: number}>}
    */
   copyableRanges = [];
+
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:197]:
```suggestion
```

File: handsontable/src/plugins/copyPaste/copyPaste.js [hunk 14]
```diff
@@ -253,7 +265,7 @@ export class CopyPaste extends BasePlugin {
   }
 
   /**
-   * Disables the plugin functionality for this Handsontable instance.
+   * Enables the [`CopyPaste`](#copypaste) plugin for your Handsontable instance.
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/copyPaste.js:268]:
```suggestion
   * Disables the [`CopyPaste`](#copypaste) plugin for your Handsontable instance.
```

**Review by budnix — CHANGES_REQUESTED** [review]

File: handsontable/src/dataMap/metaManager/metaSchema.js (hunk 1 above)
Comment (kirszenbaum) [reply handsontable/src/dataMap/metaManager/metaSchema.js]: Sure, done: [dd8ab83](https://github.com/handsontable/handsontable/pull/10118/commits/dd8ab8380c1554f3ff6339aaa7a77c2826ca0828)

File: handsontable/src/core.js [hunk 15]
```diff
@@ -3594,21 +3594,44 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
...
+   * ```js
+   * // get the contents of all column headers
+   * hot.getColHeader();
+   *
+   * // get the contents of a specific column header
+   * hot.getColHeader(5);
+   *
+   * // get the contents of all column headers at a specific level
+   * hot. getColHeader(null, -2)
+   *
```
Comment (budnix) [inline handsontable/src/core.js:3620]:
```suggestion
   * // get the contents of all column headers at a specific level
   * hot. getColHeader(null, -2)
   *
```
This syntax is not supported and should be removed from Docs.

File: handsontable/src/core.js [hunk 16]
```diff
@@ -3594,21 +3594,44 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
...
+   * Read more:
+   * - [Guides: Column groups](@/guides/columns/column-groups.md)
+   * - [Options: `colHeaders`](@/api/options.md#colheaders)
+   * - [Guides: Copy with headers](@/guides/cell-features/clipboard.md#copy-with-headers)
+   *
+   * ```js
+   * // get the contents of all column headers
+   * hot.getColHeader();
+   *
+   * // get the contents of a specific column header
```
Comment (budnix) [inline handsontable/src/core.js:3615]:
```suggestion
   * // get the contents of a specific most-bottom column header
```
Do you think it's worth adding information that you are getting (by default) the most-bottom header value?

**Review by budnix — CHANGES_REQUESTED** [review]

File: docs/content/guides/cell-features/clipboard.md [hunk 17]
```diff
@@ -110,6 +109,107 @@ ReactDOM.render(<ExampleComponent />, document.getElementById('example1'));
...
+    copyColumnGroupHeaders: true,
+    copyColumnHeadersOnly: true,
+  },
+  colHeaders: true,
+  rowHeaders: true,
+  height: 'auto',
+  nestedHeaders: [
+    ['H', { label: 'I', colspan: 2 }, { label: 'J', colspan: 2 }, { label: 'K', colspan: 2 }, { label: 'L', colspan: 2 }, 'M'],
+    ['N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W']
+  ],
```
Comment (budnix) [inline docs/content/guides/cell-features/clipboard.md:161]:
```suggestion
  nestedHeaders: [
    ['A', { label: 'B', colspan: 2 }, { label: 'C', colspan: 2 }, { label: 'D', colspan: 2 }, { label: 'E', colspan: 2 }, 'F'],
    ['G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P']
  ],
```
I'd follow the same rules as for the nested headers/collapsible columns examples where the header names start from A...Z

File: docs/content/guides/cell-features/clipboard.md [hunk 18]
```diff
@@ -110,6 +109,107 @@ ReactDOM.render(<ExampleComponent />, document.getElementById('example1'));
...
+      copyColumnGroupHeaders: true,
+      copyColumnHeadersOnly: true,
+    }}
+    colHeaders={true}
+    rowHeaders={true}
+    height="auto"
+    nestedHeaders={[
+      ['H', { label: 'I', colspan: 2 }, { label: 'J', colspan: 2 }, { label: 'K', colspan: 2 }, { label: 'L', colspan: 2 }, 'M'],
+      ['N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W']
+    ]}
```
Comment (budnix) [inline docs/content/guides/cell-features/clipboard.md:200]:
```suggestion
    nestedHeaders={[
      ['A', { label: 'B', colspan: 2 }, { label: 'C', colspan: 2 }, { label: 'D', colspan: 2 }, { label: 'E', colspan: 2 }, 'F'],
      ['G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P']
    ]}
```
...and, of course, React version.

File: handsontable/src/core.js (hunk 15 above)
Comment (kirszenbaum) [reply handsontable/src/core.js]: Done: [25e0e9e](https://github.com/handsontable/handsontable/pull/10118/commits/25e0e9e161f694c2a9cacd94e905ec8317614ad2)

File: handsontable/src/core.js (hunk 16 above)
Comment (kirszenbaum) [reply handsontable/src/core.js]: Definitely, done: [25e0e9e](https://github.com/handsontable/handsontable/pull/10118/commits/25e0e9e161f694c2a9cacd94e905ec8317614ad2)

**Review by budnix — CHANGES_REQUESTED** [review]

File: handsontable/src/pluginHooks.js [hunk 19]
```diff
@@ -1463,23 +1461,20 @@ const REGISTERED_HOOKS = [
...
-   * Since the 12.3.0 the hook is triggered with the 3rd `copiedHeadersCount` argument.
+   * Fired before values are copied to the clipboard.
    *
    * @event Hooks#beforeCopy
    * @param {Array[]} data An array of arrays which contains data to copied.
    * @param {object[]} coords An array of objects with ranges of the visual indexes (`startRow`, `startCol`, `endRow`, `endCol`)
    *                         which will copied.
-   * @param {{ columnHeadersCount: number }} copiedHeadersCount An object with keys that holds information with
-   *                                                            the number of copied headers.
+   * @param {{ columnHeadersCount: number }} copiedHeadersCount (Since 12.3.0) The number of copied column headers.
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply handsontable/src/pluginHooks.js:1477]: Done: [7bc6756](https://github.com/handsontable/handsontable/pull/10118/commits/7bc6756a15eac66f1c1b4c6fc4460967f5c29fb1)

File: docs/content/guides/cell-features/clipboard.md [hunk 20]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Clipboard
 metaTitle: Clipboard - JavaScript Data Grid | Handsontable
-description: Copy data from selected cells to the clipboard, using the "Ctrl/Cmd + C" shortcut or the right-click context menu. Control the clipboard with Handsontable's API.
+description: Copy data from selected cells to the clipboard, using the <kbd>**Ctrl**</kbd> / <kbd>**Cmd**</kbd>  + <kbd>**C**</kbd> shortcut or the context menu. Control the clipboard with Handsontable's API.
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply docs/content/guides/cell-features/clipboard.md]: Right, done: [7bc6756](https://github.com/handsontable/handsontable/pull/10118/commits/7bc6756a15eac66f1c1b4c6fc4460967f5c29fb1)

File: handsontable/src/pluginHooks.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (budnix) [reply handsontable/src/pluginHooks.js:1477]: Thanks! 🙏

**Pull request merged**